client/js/app/stores/ExplorerStore.js

```diff
@@ -100,6 +100,17 @@ function _defaultStep() {
   }
 }
 
+/**
+ * Clone the query attributes to a new object.
+ * @param {Object} source  Source object from get the data to be copied.
+ * @returns {Object} Copy of the query attributes of th source object
+ */
+function _cloneAttrs(source) {
+	return {
+	    query: source.query
```

Comment:
Quick note here: Objects are passed by reference so this actually wouldn't properly clone the query. You need to use something like lodash's `_.cloneDeep`.